docs/HowToDevelopLoadableApps.md

```diff
@@ -0,0 +1,216 @@
+3. The total ram requirement of an application needs to be fixed at build time.
+4. Currently TizenRT supports only elf binary format
+
+## Steps to enable existing loadable apps
+
+### Configuration Settings
+
+#### Essential Configurations
+
+- The following configs should be enabled:
+
+```
+ ------------------------------------------------------------------------------------------------------
+ CONFIG FLAG					VALUE		DESCRIPTION
+ ------------------------------------------------------------------------------------------------------
+
+ CONFIG_APP_BINARY_SEPARATION			y		App binary separation
+
+ CONFIG_NUM_APPS				x		x = number of apps
+
+ CONFIG_EXAMPLES_ELF				y		Loadable Apps test
+
+ CONFIG_BINARY_MANAGER				y		Binary Manager
+ CONFIG_BINMGR_RECOVERY				y		Binary Recovery Management (OPTIONAL)
+ CONFIG_BINMGR_UPDATE				y		Binary Manager Update APIs (OPTIONAL)
+
+ CONFIG_BINFMT_ENABLE				y		Binary Loader
+ CONFIG_BINFMT_LOADABLE				y		Loadable binary format
+ CONFIG_ELF					y		Enable the ELF Binary Format
+ CONFIG_ELF_ALIGN_LOG2				2		Log2 Section Alignment
+ CONFIG_ELF_STACKSIZE				2048		ELF Stack Size
+ CONFIG_ELF_BUFFERSIZE				32		ELF I/O Buffer Size
+ CONFIG_ELF_BUFFERINCR				32		ELF I/O Buffer Realloc Increment
+ CONFIG_ELF_EXCLUDE_SYMBOLS			y		Excludes symbol information from ELF file
+ CONFIG_ELF_CACHE_READ				n		ELF cache read support
+ CONFIG_SYMTAB_ORDEREDBYNAME			n		Symbol Tables Ordered by Name
+ CONFIG_OPTIMIZE_APP_RELOAD_TIME		y		Optimizations for application reload time
+ ------------------------------------------------------------------------------------------------------
+
+```
+
+#### Optional configurations
+
+These are not a part of the core loadable apps but can be optionally enabled to test some of the features.
+Make sure that the maximum task size is increased to accomodate the extra tests enabled.
+
+1. If MPU is enabled, MPU test can also be enabled as below:
+
+```
+CONFIG_EXAMPLES_MPU_TEST=y
+CONFIG_MPUTEST_KERNEL_CODE_ADDR=0x60000000
+CONFIG_MPUTEST_KERNEL_DATA_ADDR=0x20200000
+CONFIG_MPUTEST_APP_ADDR=0x80200100
+```
+   In above scenario, the configs description is as below:
+
+``` 
+CONFIG_MPUTEST_KERNEL_CODE_ADDR		This is same as FLASH address		CONFIG_FLASH_START_ADDR
+CONFIG_MPUTEST_KERNEL_DATA_ADDR		This is same as SRAM address		CONFIG_SRAM_START_ADDR
+CONFIG_MPUTEST_APP_ADDR			Application sections' address		CONFIG_RAM_REGIONx_START
+					(described below)
+```
+
+2. Other Loadable Apps test configs that could be enabled are as below:
+
+```
+#
+# Enable Test Scenarios
+#
+CONFIG_EXAMPLES_MESSAGING_TEST=y
+CONFIG_MESSAGING_TEST_REPETITION_NUM=1
+CONFIG_EXAMPLES_RECOVERY_TEST=y
+# CONFIG_ENABLE_RECOVERY_AGING_TEST is not set
+# CONFIG_EXAMPLES_BINARY_UPDATE_TEST is not set
+# CONFIG_EXAMPLES_MICOM_TIMER_TEST is not set
+CONFIG_EXAMPLES_ELF_FULLYLINKED=y
+```
+
+### Memory Settings
+
+ELF is stored in flash partition and loaded into ram at runtime and hence we need to reserve
```

Comment:
We will change the default binary space to FS (You can refer to #4172).

Reply:
This change hasn't been merged yet. I will update the doc as and when needed post the PR gets merged